Tribler/Core/Modules/restapi/channels/channels_subscription_endpoint.py

```diff
@@ -84,10 +84,8 @@ def render_PUT(self, request):
         """
         request.setHeader('Content-Type', 'text/json')
         channel_info = self.get_channel_from_db(self.cid)
-        if channel_info is None:
-            return ChannelsModifySubscriptionEndpoint.return_404(request)
 
-        if channel_info[7] == VOTE_SUBSCRIBE:
+        if channel_info is not None and channel_info[7] == VOTE_SUBSCRIBE:
```

Comment:
Ah, because it's not necessary for a channel to be discovered if you want to subscribe to it. So if someone passes you a dispersy id of the channel, you can subscribe to it, whether you have discovered that channel already or not. Dispersy takes care of that. By returning the 404 error, I make this possible (again).